Marlin/src/lcd/ultralcd.cpp

```diff
@@ -3742,22 +3742,37 @@ void kill_screen(const char* lcd_msg) {
      * "Print from SD" submenu
      *
      */
+
     #if ENABLED(SD_REPRINT_LAST_SELECTED_FILE)
       uint32_t saved_encoderPosition = 0;
-      static millis_t assume_print_finished = 0;
+
+      void lcd_reselect_last_file() {
+        // Some of this is a hack to force the screen update to work.
+        // TODO: Fix the real issue that causes this!
+        lcdDrawUpdate = LCDVIEW_CALL_REDRAW_NEXT;
+        _lcd_synchronize();
+        safe_delay(50);
+        _lcd_synchronize();
+        lcdDrawUpdate = LCDVIEW_CALL_REDRAW_NEXT;
+        drawing_screen = screen_changed = true;
+        lcd_goto_screen(lcd_sdcard_menu, saved_encoderPosition);
+        defer_return_to_status = true;
+        lcd_update();
+      }
```

Comment:
It seems this could call `lcd_goto_screen` at the very top, and only afterward force at least one screen update.